src/WebJobs.Script.Analyzers/AssemblyCache.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.Functions.Analyzers
+{
+    // Only support extensions and WebJobs core.
+    // Although extensions may refer to other dlls.
+    public class AssemblyCache
+    {
+        // Map from assembly identities to full paths
+        public static AssemblyCache Instance = new AssemblyCache();
+
+        bool _registered;
+
+        // Assembly Display Name --> Path
+        Dictionary<string, string> _map = new Dictionary<string, string>();
+
+        // Assembly Display Name --> loaded Assembly object
+        Dictionary<string, Assembly> _mapRef = new Dictionary<string, Assembly>();
+
+        const string WebJobsAssemblyName = "Microsoft.Azure.WebJobs";
```

Comment:
Do we want to put these in a constants file?

Reply:
Thanks for all the comments! This is a pretty straight port of code @MikeStall wrote in [azure/azure-webjobs-sdk ](https://github.com/azure/azure-webjobs-sdk/), so compliments go to him. :) (I added the tests.)

Good catch here; I'll move this into Constants.